db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryListOperatorNode.java

```diff
@@ -240,6 +240,20 @@ public ValueNode bindExpression(FromList fromList,
         return this;
     }
 
+    private void addCastOnRightOperandForStringToNonStringComparison() throws StandardException {
+        if (singleLeftOperand) {
```

Comment:
We already checked this before calling this function, do you want it here too to emphasize it?

Reply:
Yes. Would you rather I make that an assertion?